### PR TITLE
Only add CoverageOutputGenerator in coverage mode.

### LINKED ISSUE #8355: Unnecessary dependencies from coverage collection
This is split out from https://github.com/bazelbuild/bazel/issues/6656#issuecomment-492863029

Every `sh_test` depends on the Java and C++ toolchains because of coverage.
Example:
```
sh_test(
    name = "foo",
    srcs = ["foo.sh"],
)
```

While fixing https://github.com/bazelbuild/bazel/issues/5331 the lcov merger was added to every sh_test in https://github.com/bazelbuild/bazel/commit/8fcc07aebb677566de6c53b5ab57f124a840cf60

This pulls in a lot of unnecessary deps, see
```
bazel query  --implicit_deps  'deps(@bazel_tools//tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator:Main,1)'

@bazel_tools//tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator:Main
@bazel_tools//tools/test/CoverageOutputGenerator/java/com/google/devtools/coverageoutputgenerator:all_lcov_merger_lib
@bazel_tools//tools/launcher:launcher
@bazel_tools//tools/jdk:current_java_toolchain
@bazel_tools//tools/jdk:current_java_runtime
@bazel_tools//tools/jdk:current_host_java_runtime
@bazel_tools//tools/jdk:JacocoCoverageRunner
@bazel_tools//tools/cpp:toolchain
@bazel_tools//tools/cpp:grep-includes
```

That also means you have to install or download the JDK via the external repository to run any sh_test. We should make this conditional.

I have no clue why this depends on the `cpp:toolchain`, cc @hlopko.

### PR DESCRIPTION
**Pull request by iirina**
Fixes #8355.

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/analysis/BaseRuleClasses.java [hunk 1]
```diff
@@ -109,6 +109,12 @@ public Object getDefault(AttributeMap rule) {
         TestConfiguration.class, defaultValue, COVERAGE_REPORT_GENERATOR_CONFIGURATION_RESOLVER);
   }
 
+  public static LabelLateBoundDefault<TestConfiguration> coverageOutputGenerator(
+      Label defaultValue) {
+    return LabelLateBoundDefault.fromTargetConfiguration(
+        TestConfiguration.class, defaultValue, COVERAGE_SUPPORT_CONFIGURATION_RESOLVER);
```
[the remark this answers is not in the document]
Comment (iirina) [reply src/main/java/com/google/devtools/build/lib/analysis/BaseRuleClasses.java]: Yup, I misread in a hurry what the resolver did. Thanks!

**Comment by iirina** [comment]
I added the tests in a new `BazelShTestConfiguredTargetTest`. We already have integration tests for coverage (including for `sh_test`). There is full test coverage right now.

File: src/test/java/com/google/devtools/build/lib/rules/sh/BazelShTestConfiguredTargetTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+/** Tests for sh_test configured target. */
+@RunWith(JUnit4.class)
+public class BazelShTestConfiguredTargetTest extends BuildViewTestCase {
+  @Test
+  public void testCcTestCoverageOutputGenerator() throws Exception {
+    scratch.file(
+        "sh/test/BUILD",
+        "sh_test(name = 'foo_test',",
+        "           srcs = ['foo_test.sh'])");
```
Comment (hlopko) [inline src/test/java/com/google/devtools/build/lib/rules/sh/BazelShTestConfiguredTargetTest.java]: Nit: reformat (either single line or each attribute on a separate line). Also below. And above :))

Comment (iirina) [reply src/test/java/com/google/devtools/build/lib/rules/sh/BazelShTestConfiguredTargetTest.java]: Done.

**Pull request closed without merging**